packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -248,6 +248,126 @@ func GetBlockProofStatus(
 	}, nil
 }
 
+// BatchGetBlocksProofStatus checks whether the batch of L2 blocks still need new proofs or new contests.
+// Here are the possible status:
+// 1. No proof on chain at all.
+// 2. A valid proof has been submitted.
+// 3. An invalid proof has been submitted, and there is no valid contest.
+// 4. An invalid proof has been submitted, and there is a valid contest.
+func BatchGetBlocksProofStatus(
+	ctx context.Context,
+	cli *Client,
+	ids []*big.Int,
+	proverAddress common.Address,
+	proverSetAddress common.Address,
+) ([]*BlockProofStatus, error) {
+	ctxWithTimeout, cancel := CtxWithTimeoutOrDefault(ctx, defaultTimeout)
+	defer cancel()
+	var (
+		parentHashes = make([][32]byte, len(ids))
+		parents      = make([]*types.Header, len(ids))
+		blockIDs     = make([]uint64, len(ids))
+		result       = make([]*BlockProofStatus, len(ids))
+	)
+	// Get the local L2 parent header.
+	g, gCtx := errgroup.WithContext(ctxWithTimeout)
+	for i, id := range ids {
+		g.Go(func() error {
+			parent, err := cli.L2.HeaderByNumber(gCtx, new(big.Int).Sub(id, common.Big1))
+			if err != nil {
+				return err
+			}
+			parentHashes[i] = parent.Hash()
+			parents[i] = parent
+			blockIDs[i] = id.Uint64()
+			return nil
+		})
+	}
+	if gErr := g.Wait(); gErr != nil {
+		return nil, gErr
+	}
+
+	// Get the transition state from TaikoL1 contract.
+	transitions, err := cli.TaikoL1.GetTransitions(
+		&bind.CallOpts{Context: ctxWithTimeout},
+		blockIDs,
+		parentHashes,
+	)
+	if err != nil {
+		return nil, err
+	}
+	g, gCtx = errgroup.WithContext(ctxWithTimeout)
+	for i, transition := range transitions {
+		// No proof on chain
+		if transition.BlockHash == (common.Hash{}) {
+			result[i] = &BlockProofStatus{IsSubmitted: false, ParentHeader: parents[i]}
+			continue
+		}
+		g.Go(func() error {
+			header, err := cli.WaitL2Header(gCtx, ids[i])
```

Comment:
We can just wait for the highest header here right? no need to send requests to check and wait every block?

Reply:
Yes